docs/dev/app/listings.md

```diff
@@ -1,9 +1,388 @@
+
+**InterestForm Props**
+- `listingData`: *Listing*. The listing whose InterestForm will be rendered. The rendering of fields is dependent on the `listing_type` of `listingData`. 
+- (optional) `interpretation`: *boolean*. Default value: false. Should be `true` for Langauge Support listings. Used to distinguish between Case Interpretations and Cases. 
+
+**Example Usage**
+- Langauge Support listings: `<InterestForm listingData={listingData} interpretation />`
+- Non-Langauge Support listings: `<InterestForm listingData={listingData} />`
+
+#### How to Customize InterestForm Questions 
+**Creating a New Question**
+1. Create a new Input Component and a new state to track the input. Update useEffect to reset the state when listingData changes. 
+2. Add a new field to the `Responses` type, corresponding to this new input 
+3. In `handleInsert`, add an error check if this new field is required for certain listing types 
+4. In `handleInsert`, conditionally add the new input to `responses` to be inserted into supabase
+
+```tsx
+interface Responses {
+  start_date?: Date;
+  needs_interpreter?: boolean;
+  interest_reason: string;
+  // (2) add new fields as necessary 
+}
+
+export default function InterestForm( ... ) {
+    const [newState, setNewState] = useState('');
+    // other states ... 
+    
+    useEffect(() => {
+        // (1) Reset form fields when caseData changes
+        setNewState('');
+        // reset other states ... 
+    }, [listingData]);
+
+    const handleInsert = async () => {
+        // Error handling, check if required fields are unfilled.
+        if (
+            // other conditional checks ... 
+            other-checks
+            // (3) new error handling check 
+            || newState === ''
+        ) {
+            setMissingInfo(true);
+            return;
+        }
+
+        const responses: Responses = { interest_reason: '' };
+        if (auth && auth.userId) {
+            // (4) conditionally add newState to responses
+            const newInterest: Interest = {
+            listing_id: listingData.id,
+            listing_type:
+                listingData.listing_type === 'CASE' && interpretation
+                    ? 'CASE_INT'
+                    : listingData.listing_type,
+                user_id: auth.userId,
+                form_response: responses,
+            };
+            await upsertInterest(newInterest);
+        }
+    }
+
+    return (
+        <Styles.FormContainer>
+            <H3>Submit Interest</H3>
+            {submitted ? (
+                <Styles.EmptySpace>
+                    <P>Your submission has been received!</P>
+                </Styles.EmptySpace>
+            ) : (
+                <Flex $gap="30px" $direction="column">
+                    {/* (1) New Interest Question Input*/}
+                    {/*... Existing Questions*/}
+                </Flex>
+    )
+}
+```
 
 ## Cases
+### Frontend
```

Comment:
no need to split frontend and backend
the framework we have is that we only talk about frontend (client side), and any backend logic should either be in the "Supabase" page or "Admin Dashboard" (renamed to "Retool" in `docs` branch)

Reply:
here, you can just give the reader a brief description of how each specific listing page differ from one another, then split into categories only if necessary

Reply:
gotcha, i deleted the description of the query functions and added the comparison table at the beginning